src/components/TapBar.tsx

```diff
@@ -0,0 +1,204 @@
+import { ReactComponent as EditIcon } from '../assets/svgs/edit.svg'
+import { ReactComponent as FriendsIcon } from '../assets/svgs/friends.svg'
+import { ReactComponent as ChattingIcon } from '../assets/svgs/chatting.svg'
+import { ReactComponent as SettingIcon } from '../assets/svgs/setting.svg'
+import { ReactComponent as SearchIcon } from '../assets/svgs/search.svg'
+import { FriendList } from './FriendList'
+import { ChatList } from './ChatList'
+import { MyPage } from './MyPage'
+
+export const TapBar = () => {
+  const inputRef = useRef<HTMLInputElement>(null)
+  //탭바 상태 localStorage에 저장
+  const [isFriend, setIsFriend] = useState<boolean>(
+    localStorage.getItem('isFriend') ? localStorage.getItem('isFriend') === 'true' : true,
+  )
+  const [isChatList, setIsChatting] = useState<boolean>(
+    localStorage.getItem('isChatList') ? localStorage.getItem('isChatList') === 'true' : false,
+  )
+  const [isSetting, setIsSetting] = useState<boolean>(
+    localStorage.getItem('isSetting') ? localStorage.getItem('isSetting') === 'true' : false,
+  )
+  const [inputValue, setInputValue] = useState<string>('')
+
+  useEffect(() => {
+    localStorage.setItem('isFriend', isFriend.toString())
+    localStorage.setItem('isChatList', isChatList.toString())
+    localStorage.setItem('isSetting', isSetting.toString())
+  }, [isFriend, isChatList, isSetting])
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(e.target.value)
+  }
+
+  //inputRef설정 함수
+  const handleInputClick = (e: React.MouseEvent<HTMLInputElement>) => {
+    e.preventDefault()
+    e.stopPropagation()
+    inputRef.current?.focus()
+  }
+
+  return (
+    <>
+      <TopHeading>
+        <UserContainer>
+          <EditIcon style={{ width: '1.9rem', color: isSetting ? colors.white : colors.grey_700 }} />
+          <TitleText style={{ color: isSetting ? colors.white : colors.grey_900 }}>
+            {isFriend ? '친구들' : isSetting ? '내 설정' : '채팅'}
+          </TitleText>
+          <WriteIcon style={{ color: isSetting ? 'transparent' : colors.grey_900, width: '1.5rem' }} />
+        </UserContainer>
+      </TopHeading>
+      {!isSetting ? (
+        <>
+          <InputContainer>
+            <InputLine>
+              <SearchIcon style={{ width: '1.5rem' }} />
+              <InputBox
+                placeholder="검색"
+                ref={inputRef}
+                value={inputValue}
+                onChange={onChange}
+                onClick={handleInputClick}
+              />
+            </InputLine>
+          </InputContainer>
+          {isFriend ? <FriendList searchValue={inputValue} /> : <ChatList searchValue={inputValue} />}
+        </>
+      ) : (
+        <MyPage />
+      )}
+
+      <TapContainer>
+        <IconBox
+          onClick={() => {
+            setIsChatting(true)
+            setIsFriend(false)
+            setIsSetting(false)
+          }}
+        >
```

Comment:
현재는 tabBarState의 아이콘마다 상태 관리를 하고 있어서 하나의 아이콘 클릭시 나머지 아이콘의 상태를 false 로 만들어줘야 해서 setState 함수를 3번 호출하고 있는데 tabBarState 라는 변수에 타입을 "isChat" | "isFriend" | "isSetting" 이렇게 string 으로 지정해주고 예를 들어 채팅 아이콘을 클릭하면 해당 상태로 바뀌는 로직이 한번만 실행될 수 있게 짜셔도 좋을 것 같아요!!

```suggestion

const enum TABBAR_STATE{
   'IS_CHAT', 'IS_FRIEND', 'IS_SETTING'
}

const [tabBarState,setTabBarState]=useState<number>(TABBAR_STATE.IS_CHAT)

...

<IconBox
          onClick={() => {
            setTabBarState(TABBAR_STATE.IS_CHAT)
          }}
        >

...
<IconBox
          onClick={() => {
            setTabBarState(TABBAR_STATE.IS_FRIEND)
          }}
        >

...

<IconBox
          onClick={() => {
            setTabBarState(TABBAR_STATE.IS_SETTING)
          }}
        >

...

Reply:
다음에는 꼭 이렇게...!! 해보겠습니다